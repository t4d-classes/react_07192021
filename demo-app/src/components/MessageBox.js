import { Component } from 'react';
import PropTypes from 'prop-types';

export class MessageBox extends Component {

  static get propTypes() {
    return {
      headerText: PropTypes.string.isRequired,
      children: PropTypes.node.isRequired,
    };
  }

  render() {

    return (
      <div className="panel panel-default">
        <h3 className="panel-heading">
          {this.props.headerText}
        </h3>
        <div className="panel-body">
          {this.props.children}
        </div>
      </div>
    );

  }


}