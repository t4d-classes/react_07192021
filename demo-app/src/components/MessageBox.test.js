import React from "react";

import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

import { MessageBox } from "./MessageBox";

test("snapshot MessageBox component", () => {
  expect(
    renderer.create(
      <MessageBox headerText="Info">
        Some Content
      </MessageBox>
    ).toJSON()
  ).toMatchSnapshot();
});

describe("MessageBox component", () => {
  test("renders MessageBox component", () => {
    const { getByText, getByRole } = render(
      <MessageBox headerText="Info">
        Some Content
      </MessageBox>
    );

    // this is ok
    expect(getByText("Info")).toBeInTheDocument();
    expect(getByText("Some Content")).toBeInTheDocument();

    // but this is better
    expect(getByRole("heading").textContent).toBe("Info");
    expect(getByRole("complementary").textContent).toBe("Some Content");
  });
});
