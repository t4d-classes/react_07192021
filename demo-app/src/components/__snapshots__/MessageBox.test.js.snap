// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot MessageBox component 1`] = `
<div
  className="panel panel-default"
>
  <h3
    className="panel-heading"
  >
    Info
  </h3>
  <div
    className="panel-body"
    role="complementary"
  >
    Some Content
  </div>
</div>
`;
